contracts/tasks/helpers/Deployment.ts

```diff
@@ -341,11 +342,11 @@ export class Deployment {
    * @returns deployed contract
    */
   async deployContractWithLinkedLibraries<T extends BaseContract>(
-    contractFactory: ContractFactory,
+    { contractFactory, signer }: IDeployContractWithLinkedLibrariesArgs,
     ...args: unknown[]
   ): Promise<T> {
-    const deployer = await this.getDeployer();
-    const feeData = await deployer.provider.getFeeData();
+    const deployer = signer || (await this.getDeployer());
+    const feeData = await deployer.provider!.getFeeData();
```

Comment:
It's better to use `?.` here and `feeData?.maxFeePerGas`, `feeData?.maxPriorityFeePerGas,`,

Reply:
Sure, I just made sure that `maxFeePerGas` and `maxPriorityFeePerGas` could be undefined. Updated.